lib/modules/dosomething/dosomething_reportback/includes/dosomething_reportback_file.inc

```diff
@@ -66,7 +66,7 @@ class ReportbackFileEntity extends Entity {
         $reason = $values['flagged_reason'];
       }
       $reportback->flag($reason);
-      if (isset($values['delete'])) {
+      if (isset($values['delete']) && $values['delete']) {
```

Comment:
Isn't `isset($values['delete']) && $values['delete']` is the same as `!empty($values['delete'])`?